src/enums/ClaimConditionType.ts

```diff
@@ -0,0 +1,27 @@
+enum ClaimConditionType {
+  StartTime,
+  LimitPerWallet,
+  MerkleRoot,
+  WaitTimeBetweenClaims,
+}
+
+/**
+ * Questions:
+ *
```

Comment:
high level, contract specific details:
mint conditions = set the conditions on who can mint the nft.
each condition can be thought of each mint stage with different properties.

there's no limit on how many conditions can be set on the contract. 
the list of conditions has to be sorted by `startTimestamp` in an ascending order and must be unique. 
for example, a list of 3 condtions: [startTimestamp = 0, startTimestamp=30, startTimestamp=60].